src/sentry/static/sentry/app/views/stream/tagFilter.jsx

```diff
@@ -28,103 +32,143 @@ class StreamTagFilter extends React.Component {
     super(...args);
     this.state = {
       query: '',
-      loading: false,
+      isLoading: false,
       value: this.props.value,
+      textValue: this.props.value,
     };
-  }
-
-  componentDidMount() {
-    let select = this.refs.select;
-
-    let selectOpts = {
-      placeholder: '--',
-      allowClear: true,
-    };
-
-    if (!this.props.tag.predefined) {
-      Object.assign(selectOpts, {
-        initSelection: (element, callback) => {
-          callback(StreamTagFilter.tagValueToSelect2Format(this.props.value));
-        },
-        ajax: {
-          url: this.getTagValuesAPIEndpoint(),
-          dataType: 'json',
-          delay: 250,
-          data: (term, page) => {
-            return {
-              query: term,
-            };
-          },
-          results: (data, page) => {
-            // parse the results into the format expected by Select2
-            return {
-              results: _.map(data, val =>
-                StreamTagFilter.tagValueToSelect2Format(val.value)
-              ),
-            };
-          },
-          cache: true,
-        },
-      });
-    }
-
-    $(select)
-      .select2(selectOpts)
-      .select2('val', this.state.value)
-      .on('change', this.onSelectValue);
+    this.api = new Client();
   }
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.value !== this.state.value) {
-      this.setState(
-        {
-          value: nextProps.value,
-        },
-        () => {
-          let select = this.refs.select;
-          $(select).select2('val', this.state.value);
-        }
-      );
+      this.setState({
+        value: nextProps.value,
+        textValue: nextProps.value,
+      });
     }
   }
 
   componentWillUnmount() {
-    let select = ReactDOM.findDOMNode(this.refs.select);
-    $(select).select2('destroy');
+    if (!this.api) return;
+    this.api.clear();
+    this.api = null;
```

Comment:
How come we're manually cleaning this up?

Reply:
Because it's generally done in ApiMixin, we don't have a HoC for it.

Reply:
oh i mean the line `this.api = null`, not the part about clearing outstanding requests

Reply:
Ah yeah, it's probably un-needed but doesn't hurt to explicitly clear out references

Reply:
I don't think we should do it, unless consistently everywhere (which i've not seen)

Reply:
Why not, what's the harm?

Reply:
1 - it's a line of code that doesn't do anything, you could argue that there's never any harm in those, but there's also no point
2 - in this case it's inconsistent practice with the rest of our codebase

Reply:
I think it's better to be a bit proactive with cleaning up your references but I've removed it